### PR TITLE
WebP for background-image inline style

### PR DESCRIPTION
**Pull request by DevIntact**
#48

### CODE REVIEW

File: inc/cache_enabler_disk.class.php [hunk 1]
```diff
@@ -283,12 +283,17 @@ private static function _create_files($data) {
...
-            // call the webp converter callback
-            $converted_data = preg_replace_callback($regex_rule,'self::_convert_webp',$data);
-
+            // magic regex rules
+            $regex_rules = array(
+				'#(?<=(?:(ref|src|set)=[\"\']))(?:http[s]?[^\"\']+)(\.png|\.jp[e]?g)(?:[^\"\']+)?(?=[\"\')])#',
+                '#(?<=(?:(url)))\((?![\'\"]?(?:data):)[\'\"]?([^\'\"\)]*)[\'\"]?\)#'
+			);
+            
+			$converted_data = $data;
```
Comment (svenba) [inline inc/cache_enabler_disk.class.php]: Odd formatting.

Comment (DevIntact) [reply inc/cache_enabler_disk.class.php]: commit 7b9f9c029104a24ba09e0e40952b08b485d78230

**Comment by svenba** [comment]
While I understand to have as much as possible converted to webp URLs. However, having inline-CSS is not recommended for clean web development.

**Comment by hebbet** [comment]
many wordpress theme add slider images for example as inline background.

**Comment by svenba** [comment]
We'll add a filter at this position. So you can define what you need.

**Pull request closed without merging**